internal/internal_workflow_client.go

```diff
@@ -877,6 +927,69 @@ func (wc *WorkflowClient) ResetWorkflowExecution(ctx context.Context, request *w
 	return resp, nil
 }
 
+func (wc *WorkflowClient) UpdateWorkflowWithOptions(
+	ctx context.Context,
+	req *UpdateWorkflowWithOptionsRequest,
+) (WorkflowUpdateHandle, error) {
+	if err := wc.ensureInitialized(); err != nil {
+		return nil, err
+	}
+	argPayloads, err := wc.dataConverter.ToPayloads(req.Args...)
+	if err != nil {
+		return nil, err
+	}
+	header, _ := headerPropagated(ctx, wc.contextPropagators)
+
+	grpcCtx, cancel := newGRPCContext(ctx, defaultGrpcRetryParameters(ctx))
+	defer cancel()
+	resp, err := wc.workflowService.UpdateWorkflow(grpcCtx, &workflowservice.UpdateWorkflowRequest{
```

Comment:
I think we made a mistake having query interceptor call query with options. All client query/update calls should go through the interceptor IMO. I opened https://github.com/temporalio/sdk-go/issues/991 for this just now on queries. For this, we should go ahead and move this logic to inside the interceptor and call the interceptor from here and just call this overload from the non-options overload.

Reply:
Mentioned elsewhere but yes, I just pushed this to the PR.